src/views/components/map/Map.web.js

```diff
@@ -1,16 +1,510 @@
 import React, { Component } from 'react';
-import { Box, Text } from '../../components';
+import { string, object, array, number, bool, any, func } from 'prop-types';
+import config from '../../../config';
 
+/* global google */
 class Map extends Component {
-  render() {
-    return (
-      <Box>
-        <Text>
-          TODO Map
-        </Text>
-      </Box>
-    );
-  }
+    static defaultProps = {
+      controls: false,
+      zoom: 12,
+      center: { lat: -33.8688, lng: 151.2093 },
+      markers: [],
+      routes: [],
+      icon: 'https://i.imgur.com/unMXE8B.png',
+      iconClick: 'https://i.imgur.com/XiZYxed.png',
+      mapRouteIcon: 'https://i.imgur.com/Zyinht5.png',
+      suppressMarkers: true,
+      polygons: [],
+      polylines: [],
+      height: 300,
+      width: 400,
+    }
+
+    static propTypes = {
+      style: object,
+      suppressMarkers: bool,
+      controls: bool,
+      zoom: number,
+      maxZoom: number,
+      center: any,
+      markers: array,
+      routes: array,
+      polygons: array,
+      polylines: array,
+      icon: string,
+      iconClick: string,
+      lat: any,
+      lng: any,
+      onClick: func,
+      mapRouteIcon: string,
+      height: string, 
+      width: string,
+
+    }
+
+    state = {
+      error: null,
+    };
+
+    UNSAFE_componentWillMount() { 
+      this.setupGoogleApi();
+    }
+    
+    componentWillUnmount() {
+      if ( this.map ) {
+        this.map = null;
+      }
+    }
+
+    setupGoogleApi = () => {
```

Comment:
We already do this in GoogleProvider - are we able to reuse that function / remove this as it shouldn't be needed? I want to try eliminate duplicated code.